File: 240415/이동경로상에 있는 모든 숫자 더하기/add-all-the-numbers-on-the-path.py
N, T = map(int, input().split())
commands = input()
board = [list(map(int, input().split())) for _ in range(N)]

move_dirs = {0: (-1, 0), 1: (0, 1), 2: (1, 0), 3: (0, -1)}
x, y = len(board) // 2, len(board[0]) // 2
dir_num = 0

answer = board[x][y]
for cmd in commands:
    if cmd == 'R':
        dir_num = (dir_num + 1) % 4
    elif cmd == 'L':
        dir_num = (dir_num + 3) % 4
    else:
        if 0 <= x + move_dirs[dir_num][0] < N and 0 <= y + move_dirs[dir_num][1] < N:
            x += move_dirs[dir_num][0]
            y += move_dirs[dir_num][1]
            answer += board[x][y]
        else:
            continue

print(answer)

File: 260207/이동경로상에 있는 모든 숫자 더하기/add-all-the-numbers-on-the-path.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int N, T;
    cin >> N >> T;

    string commands;
    cin >> commands;

    vector<vector<long long>> board(N, vector<long long>(N));
    for (int i = 0; i < N; i++) 
    {
        for (int j = 0; j < N; j++) cin >> board[i][j];
    }

    int dx[4] = {-1, 0, 1, 0};
    int dy[4] = {0, 1, 0, -1};

    int x = N / 2, y = N / 2;
    int dir = 0;

    long long answer = board[x][y];

    for (char cmd : commands) 
    {
        if (cmd == 'R') 
        {
            dir = (dir + 1) % 4;
        } 
        else if (cmd == 'L') 
        {
            dir = (dir + 3) % 4;
        } else 
        {
            int nx = x + dx[dir];
            int ny = y + dy[dir];
            if (0 <= nx && nx < N && 0 <= ny && ny < N) 
            {
                x = nx; y = ny;
                answer += board[x][y];
            }
        }
    }

    cout << answer << "\n";
    return 0;
}